maintenance/importConfig.ts

```diff
@@ -251,41 +335,35 @@ async function parseOzwProduct(
 	const productName = product.name;
 
 	// for some reasons some products already have the prefix `0x`, remove it
-	product.id = product.id.replace("0x", "");
-	product.type = product.type.replace("0x", "");
+	product.id = product.id.replace(/^0x/, "");
+	product.type = product.type.replace(/^0x/, "");
 
-	const productId = "0x" + padStart(product.id, 4);
-	const productType = "0x" + padStart(product.type, 4);
+	const productId = formatId(product.id);
+	const productType = formatId(product.type);
 
 	const fileName = `${manufacturerId}/${labelToFilename(productLabel)}.json`;
 	const filePath = path.join(processedDir, fileName);
 
-	let existingDevice;
-
-	try {
-		existingDevice = JSON5.parse(await fs.readFile(filePath, "utf8"));
-	} catch (error) {
-		// device doesn't exists
-	}
-
-	// @ts-ignore
-	const json: Record<string, any> = xml2json.toJson(productFile, {
+	let json: Record<string, any> = xml2json.toJson(productFile, {
 		object: true,
-		coerce: true,
-	}).Product;
+		coerce: true, // coerce types
+	});
+
+	json = json.Product;
 
-	// let metadata = json.MetaData?.MetaDataItem || [];
-	// metadata = Array.isArray(metadata) ? metadata : [metadata];
+	// const metadata = ensureArray(json.MetaData?.MetaDataItem);
 	// const name = metadata.find((m: any) => m.name === "Name")?.$t;
 	// const description = metadata.find((m: any) => m.name === "Description")?.$t;
 
 	if (
-		(await lookupDevice(
+		!(await lookupDevice(
 			manufacturerIdInt,
 			parseInt(product.type, 16),
 			parseInt(product.id, 16),
-		)) === undefined
+		))
 	) {
+		// let this here, if program.index is false it will not
+		// write the device index to file
```

Comment:
couldn't you just add `program.index &&` in front of the if-condition?

Reply:
Yes but in case of devices this would make the lookup at the end to fail :)

Reply:
Ohhh - I didn't make that connection in my head